tiledb/api/src/datatype/mod.rs

```diff
@@ -517,211 +523,184 @@ impl TryFrom<ffi::tiledb_datatype_t> for Datatype {
-        type Datatype = $crate::Datatype;
-        match $datatype {
-            Datatype::Int8 => { type $typename = i8; $then },
-            Datatype::Int16 => { type $typename = i16; $then },
-            Datatype::Int32 => { type $typename = i32; $then },
-            Datatype::Int64 => { type $typename = i64; $then },
-            Datatype::UInt8 => { type $typename = u8; $then },
-            Datatype::UInt16 => { type $typename = u16; $then },
-            Datatype::UInt32 => { type $typename = u32; $then },
-            Datatype::UInt64 => { type $typename = u64; $then },
-            Datatype::Float32 => { type $typename = f32; $then },
-            Datatype::Float64 => { type $typename = f64; $then },
-            Datatype::Char => unimplemented!(),
-            Datatype::StringAscii => unimplemented!(),
-            Datatype::StringUtf8 => unimplemented!(),
-            Datatype::StringUtf16 => unimplemented!(),
-            Datatype::StringUtf32 => unimplemented!(),
-            Datatype::StringUcs2 => unimplemented!(),
-            Datatype::StringUcs4 => unimplemented!(),
-            Datatype::Any => unimplemented!(),
-            Datatype::DateTimeYear => unimplemented!(),
-            Datatype::DateTimeMonth => unimplemented!(),
-            Datatype::DateTimeWeek => unimplemented!(),
-            Datatype::DateTimeDay => unimplemented!(),
-            Datatype::DateTimeHour => unimplemented!(),
-            Datatype::DateTimeMinute => unimplemented!(),
-            Datatype::DateTimeSecond => unimplemented!(),
-            Datatype::DateTimeMillisecond => unimplemented!(),
-            Datatype::DateTimeMicrosecond => unimplemented!(),
-            Datatype::DateTimeNanosecond => unimplemented!(),
-            Datatype::DateTimePicosecond => unimplemented!(),
-            Datatype::DateTimeFemtosecond => unimplemented!(),
-            Datatype::DateTimeAttosecond => unimplemented!(),
-            Datatype::TimeHour => unimplemented!(),
-            Datatype::TimeMinute => unimplemented!(),
-            Datatype::TimeSecond => unimplemented!(),
-            Datatype::TimeMillisecond => unimplemented!(),
-            Datatype::TimeMicrosecond => unimplemented!(),
-            Datatype::TimeNanosecond => unimplemented!(),
-            Datatype::TimePicosecond => unimplemented!(),
-            Datatype::TimeFemtosecond => unimplemented!(),
-            Datatype::TimeAttosecond => unimplemented!(),
-            Datatype::Blob => unimplemented!(),
-            Datatype::Boolean => unimplemented!(),
-            Datatype::GeometryWkb => unimplemented!(),
-            Datatype::GeometryWkt => unimplemented!(),
-        }
-    }};
-
-    ($func:ident, $datatype:expr$(, $arg:expr)* => $then:expr) => {{
         type Datatype = $crate::Datatype;
         match $datatype {
             Datatype::Int8 => {
-                let $func = $func::<i8>($($arg,)*);
+                type $typename = $crate::datatype::Int8Type;
```

Comment:
This all looks great, though I would have a slight preference for using fully qualified names instead of the re-exports. Feel free to not bother here though. It'll be a straightforward mechanical change if/when we get around to clarifying our public symbols and so on.

Reply:
You're definitely right about this, updated.